### ISSUE TITLE
Questions about the loss used for optimizing the proxy model

### ISSUE THREAD

**Issue by clarkkent0618** [sangmichaelxie/doremi#25]
@sangmichaelxie It seems that the loss used for optimizing the proxy model in the code is different from the one described in the paper.
```py
loss = (pertoken_loss * curr_domain_weights.detach()).sum() / normalizer
```

In the code, you directly use the proxy model's own loss here to optimize. But in the paper, the loss seems to be the minimax loss which uses the excess loss. So which one should I conform? Or there is something wrong with my understanding. Thanks.

<img width="807" alt="image" src="https://github.com/sangmichaelxie/doremi/assets/33905626/83158ca9-eef3-4982-8219-a0c7f8181051">

**Comment by yuzc19** [comment]
I have another question. When training the main model, what is the difference between resampling the data from the new distribution or using new weights to re-weight loss? Will these two have a significant performance gap?
<img width="879" alt="Screenshot 2024-01-16 at 12 20 24" src="https://github.com/sangmichaelxie/doremi/assets/53684662/a47d8f52-adfb-4277-8d09-739abb1583cc">

**Comment by sangmichaelxie** [comment]
> In the code, you directly use the proxy model's own loss here to optimize. But in the paper, the loss seems to be the minimax loss which uses the excess loss. So which one should I conform? Or there is something wrong with my understanding. Thanks.

The reference model loss is a constant with respect to the proxy model's parameters, so it doesn't affect the proxy model update and we omit it. The reference model loss does affect the domain weight update.

> When training the main model, what is the difference between resampling the data from the new distribution or using new weights to re-weight loss? Will these two have a significant performance gap?

Check out the main paragraph in pg 9 and Table 3b in the paper.

**Comment by yuzc19** [comment]
I checked it, and it makes sense. Thank you!
